Core/Src/torque_manager.c

```diff
@@ -110,6 +118,28 @@ void rampa_torque_lateral(lateral_t *ref_torque_decrease, uint32_t *ref_torque)
     }
 }
 
+void rampa_torque_longitudinal(longitudinal_t *ref_torque_decrease, uint32_t *ref_torque) {
+    static uint32_t ref_torque_ant[2] = {0, 0};
+    ref_torque[R_MOTOR] = (uint32_t)((float)(get_global_var_value(SELECTED_MODE).torq_gain * get_global_var_value(THROTTLE_PERCENT)) / 10);
+    ref_torque[L_MOTOR] = (uint32_t)((float)(get_global_var_value(SELECTED_MODE).torq_gain * get_global_var_value(THROTTLE_PERCENT)) / 10);
+
+    ref_torque[R_MOTOR] = (uint32_t)((float)ref_torque[R_MOTOR] - ref_torque_decrease->ref_decrease_R);
+    ref_torque[L_MOTOR] = (uint32_t)((float)ref_torque[L_MOTOR] - ref_torque_decrease->ref_decrease_L);
```

Comment:
Olha esse mesmo código no controle lateral, lá tem um max() que impede de ref_torque ser negativa (se `ref_torque_decrease->ref_decrease_x` for maior que `ref_torque[X_MOTOR]`)
Mas ainda nessa função, ela tá muito parecida com a `rampa_torque_lateral`. Junta as duas, talvez vc tenha que modificar o controle lateral, mas vai ser melhor.